tests/Test/AbstractConfigTestCaseTest.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Test;
+
+use PhpCsFixer\Config;
+use PhpCsFixer\Fixer\DeprecatedFixerInterface;
+use PhpCsFixer\Fixer\FixerInterface;
+use PhpCsFixer\FixerFactory;
+use PhpCsFixer\RuleSet;
+use PhpCsFixer\Test\AbstractConfigTestCase;
+use PHPUnit\Framework\ExpectationFailedException;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Test\AbstractConfigTestCase
+ */
+final class AbstractConfigTestCaseTest extends AbstractConfigTestCase
```

Comment:
so, this class is testing `AbstractConfigTestCase` helper, and same time it inherits from it ?

Reply:
> so, this class is testing `AbstractConfigTestCase` helper, and same time it inherits from it ?

Exactly, pretty much like
https://github.com/sebastianbergmann/phpunit/blob/e4390fae89bf940f430604f6fd4b58804a530816/tests/unit/Framework/AssertTest.php#L27-L36

> shouldn't `AbstractConfigTestCase` be used by `ConfigTest`,

`ConfigTest` tests what a Config is and how a Config behaves, `AbstractConfigTestCase` tests how much a used and configured Config adhere to the standards where are establishing here.

>  same as `FooFixerTest extends AbstractFixerTestCase` ?

I don't get this suggestion: in this case the tested class would be `FooFixer`, but here I need to test (in this comparison) `AbstractFixerTestCase`.
Why? Because, unlike `AbstractFixerTestCase`, `AbstractConfigTestCase` will not be `@internal` as it is intended to be used by the final user, and for the same reason it's placed under the `src/` folder.